SU2_CFD/include/variables/CMeshVariable.hpp

```diff
@@ -93,9 +93,11 @@ class CMeshVariable : public CVariable {
   /*!
    * \brief Recover the value of the adjoint of the mesh coordinates.
    */
-  inline void GetAdjoint_MeshCoord(unsigned long iPoint, su2double *adj_mesh) const final {
-    for (unsigned long iDim = 0; iDim < nDim; iDim++)
-      adj_mesh[iDim] = SU2_TYPE::GetDerivative(Mesh_Coord(iPoint,iDim));
+  inline void GetAdjoint_MeshCoord(unsigned long iPoint, su2double *adj_mesh) final {
+    for (unsigned long iDim = 0; iDim < nDim; iDim++) {
+      adj_mesh[iDim] = AD::GetDerivative(AD_InputIndex(iPoint,iDim));
+      AD::ResetInput(Mesh_Coord(iPoint,iDim));
```

Comment:
I think the trend is that if something is an input the index is not cleared by the dummy recording (since nothing is written to an input) and so you have to clear it explicitly.
Now, I guess this is only strictly required if sometimes (i.e. some recordings) you register that variable and sometimes you don't. If you always register it then the index is always valid, but if you leave it with "dangling" indices then you have a problem.
From my comment above it also seems that the location where this reset is performed matters... (but I might also have changed two things at the same time while testing).